### PR TITLE
ci: publish binaries on release

### PR DESCRIPTION
**Pull request by tiagolobocastro**
When a release is created, automatically download and upload artifacts to the release.

## Description

Publish artifacts to the release in their tar format.

## Motivation and Context

Automatically add the release artifacts to the release.
Also ensures that the artifacts are uploaded as a tar file rather than a zip containing a tar.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.

### CODE REVIEW

**Review by pchandra19 — APPROVED** [review]

File: scripts/github/release-artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die() {
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
```
Comment (niladrih) [inline scripts/github/release-artifacts.sh:31]: If you want, you can use the functions from here: https://github.com/openebs/mayastor-extensions/blob/develop/scripts/utils/log.sh

The echo_stderr is log_to_stderr
The die is log_fatal

**Review by niladrih — APPROVED** [review]
Left a minor comment

**Comment by tiagolobocastro** [comment]
@reitermarkus this will ensure we upload the tar file to the release and not the zip file (seems we always get a zip file when downloading from workflows, there's no way around it)
I'll talk to the team and see if there's scope to release 2.4.1 with a few things we've fixed (including this), and then you may update your krew PR to that which will have the LICENSE as required.

**Comment by tiagolobocastro** [comment]
@dsharma-dc responding to your comment here since there's no PR open for the other repos now.
Perhaps we can include a checksum file in the github actions artifacts, then after downloading we can compare it. I'll raise a ticket for that.

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**